src/mainwindow.cpp

```diff
@@ -4960,20 +4949,15 @@ void MainWindow::deleteSavegame_clicked()
   int count = 0;
 
   for (const QModelIndex &idx : ui->savegameList->selectionModel()->selectedIndexes()) {
-    QString name = idx.data(Qt::UserRole).toString();
+
+    auto& saveGame = m_SaveGames[idx.row()];
```

Comment:
If I'm reading this right you are sorting the vector `m_SaveGames` by filetime then adding the entries in order to the widget. That is what makes the `row()` number match the vector index. I don't really like this approach as it means that the view is directly reflecting the model but all the coupling is done by hand. If we ever want to allow sorting the with a proxy or something similar then this wouldn't work anymore.

Basically currently the code assumes that no one is going to reorder the vector and no one is going to change the view, or, in case that was to happen you would need control it so the indexes remain the same.

An alternative is to save the indexes of the vector in the `Qt::UserRole` of the row item. This way there is some decoupling and the view could be sorted by through a proxy without having to also sort the model.

Reply:
I agree, but that's probably better in a separate PR with proper display of saves ;)

Reply:
I concede to that